gguf-py/gguf/constants.py

```diff
@@ -0,0 +1,469 @@
+from __future__ import annotations
+
+import sys
+from enum import Enum, IntEnum, auto
+from typing import Any, NamedTuple
+
+#
+# constants
+#
+
+GGUF_MAGIC             = 0x46554747 # "GGUF"
+GGUF_VERSION           = 3
+GGUF_DEFAULT_ALIGNMENT = 32
+
+#
+# metadata keys
+#
+
+class GeneralKeys(NamedTuple):
+    ARCHITECTURE         = "general.architecture"
+    QUANTIZATION_VERSION = "general.quantization_version"
+    ALIGNMENT            = "general.alignment"
+    NAME                 = "general.name"
+    AUTHOR               = "general.author"
+    URL                  = "general.url"
+    DESCRIPTION          = "general.description"
+    LICENSE              = "general.license"
+    SOURCE_URL           = "general.source.url"
+    SOURCE_HF_REPO       = "general.source.huggingface.repository"
+    FILE_TYPE            = "general.file_type"
+
```

Comment:
What are you trying to do here? NamedTuple is like a tuple (i.e., an immutable collection with values you commonly access by index/position)  but with the bonus of having names for its fixed set of fields. I think you want StrEnum.